catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java

```diff
@@ -517,6 +520,164 @@ public void init() {
+         * If the call does not set the isVirtual flag, then it defaults to 'false'.
+         */
+        get("/notes/:parentid", (req, res) -> {
+            LOGGER.trace("Hit notes GET method");
+            String parentID = req.params(":parentid");
+            String[] isVirtualParams = req.queryMap()
+                    .toMap()
+                    .getOrDefault("isVirtual", new String[] {"false"});
+            Boolean isVirtual =
+                    isVirtualParams.length > 0 && Boolean.parseBoolean(isVirtualParams[0]);
+            if (!isVirtual) {
+                //Test if parent metacard ID is valid
+                try {
+                    util.getMetacard(parentID, true);
+                } catch (NotFoundException e) {
+                    LOGGER.debug("Parent metacard ID is invalid!");
+                    throw new IllegalStateException("Parent metacard ID is invalid!");
+                }
+            }
+            List<Metacard> retrievedMetacards =
+                    util.getAssociatedMetacardsByTwoAttributes(Note.PARENT_ID,
+                            Core.METACARD_TAGS,
+                            parentID,
+                            "note");
+            if (!retrievedMetacards.isEmpty()) {
+                ArrayList<String> getResponse = new ArrayList<>();
+                //Reorder in terms of date created using timsort
+                retrievedMetacards.sort(Comparator.comparing(Metacard::getCreatedDate));
+                for (Metacard metacard : retrievedMetacards) {
+                    String jsonStringToAppend = util.getResponseNoteJson(metacard);
+                    if (jsonStringToAppend != null) {
+                        LOGGER.trace("jsonStringToAppend: " + jsonStringToAppend);
+                        getResponse.add(jsonStringToAppend);
+                    }
+                }
+                LOGGER.trace("getResponse: " + getResponse.toString());
+                LOGGER.trace(
+                        "output: " + util.getResponseWrapper("success", getResponse.toString()));
+                return util.getResponseWrapper("success", getResponse.toString());
+            } else {
+                return util.getResponseWrapper("error",
+                        "Could not find any notes for this result!");
+            }
+        });
+
+        put("/notes/:id", APPLICATION_JSON, (req, res) -> {
+            LOGGER.trace("Hit notes PUT method");
+            Map<String, Object> incoming = JsonFactory.create()
+                    .parser()
+                    .parseMap(req.body());
+            String noteMetacardId = req.params(":id");
+            String note = incoming.get("note")
+                    .toString();
+            Metacard metacard;
+            try {
+                metacard = util.getMetacard(noteMetacardId);
+            } catch (NotFoundException e) {
+                throw new IllegalStateException("Note metacard ID is invalid!");
```

Comment:
Might be worth logging `noteMetacardId` here.  Is the ID really invalid here?  Or is it just the case that it wasn't found?